portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,10 +45,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       PreparedQuery results = datastore.prepare(query);
+
+      int size = results.countEntities(FetchOptions.Builder.withDefaults());
       int offset = page * numComments;
+      if (offset >= size) {
+          page--;
+          offset = page * numComments;
+      }
```

Comment:
This would work if you are one above the expected page, but not if you are two above it. What if you did something like `page = min(page,math.Floor(size/numComments))`.

Also would the UI know to stop if we keep showing them content as if they are on the last page even when they are past the last page?